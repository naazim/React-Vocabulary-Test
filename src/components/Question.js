import React from 'react';

const Question = (props) => {
    const handleNext = (e) => {
        e.preventDefault();
        console.log('target', e.target);
        const enteredWord = document.querySelector('.question__input').value.trim();

        if (enteredWord !== '') {
            props.handleNext(enteredWord);
            document.querySelector('.question__input').value = '';
        }
    }
    return (
        <div>
            <div className="question__content">
                <h2 className="question__title">{props.count + 1}</h2>
                <div className="question__box">
                    <div className="question__text">{props.word.native}</div>
                    <input
                        className="question__input"
                        type="text"
                        placeholder="Enter Foreign Word"
                    />
                </div>
            </div>
            <div className="question__footer">
                {props.count > 0 && <button className="question__button question__button-previous">ðŸ¡¸</button>}
                <button className="question__button question__button-next" onClick={handleNext}>ðŸ¡º</button>
            </div>
        </div>
    )
};

export default Question;