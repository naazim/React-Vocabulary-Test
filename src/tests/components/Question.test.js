import React from 'react';
import { shallow } from 'enzyme';
import Question from '../../components/Question';

describe('Question', () => {
    const handleNextSpy = jest.fn();
    const word = {"native":"Good","foreign":"Gut"};
    const wrapper = shallow(<Question word={word} handleNext={handleNextSpy} />);

    test('should render Question correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});