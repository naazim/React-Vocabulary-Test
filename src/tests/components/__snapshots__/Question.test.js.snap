// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should render Question correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="question__content"
  >
    <h2
      className="question__title"
    >
      NaN
    </h2>
    <div
      className="question__box"
    >
      <div
        className="question__text"
      >
        Good
      </div>
      <input
        autoFocus={true}
        className="question__input"
        placeholder="Enter Foreign Word"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="question__footer"
  >
    <button
      className="question__button question__button-next"
      type="submit"
    >
      ðŸ¡º
    </button>
  </div>
</form>
`;
